### PR TITLE
Adds generation of attestation proofs

### PR DESCRIPTION
**Pull request by sudoFerraz**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/app/(dashboard)/attestation/attestationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      createAttestation(
+        {
+          address: "0xB5E5559C6b85e8e867405bFFf3D15f59693eBE2f",
+          privateData: attestationPrivateData,
+          signer: signer
+        }
+      ).then((attestationUuid) => {
+        setAttestationUuid(attestationUuid.attestationUuid)
+        const proof = createProofs(attestationPrivateData, [userLogin]);
+        console.log(JSON.stringify(proof));
```
Comment (AlecErasmus) [inline src/app/(dashboard)/attestation/attestationModal.tsx:71]: Note to remove

File: src/app/(dashboard)/attestation/attestationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      });
+    }
+  }, [signer, userAddress, attestationPrivateData]);
+
+  const handleFetchAttestationPrivateData = async () => {
+    const { data } = await axios.get("/api/attestations?team_id=" + props.teamId);
+    console.log(data);
```
Comment (AlecErasmus) [inline src/app/(dashboard)/attestation/attestationModal.tsx]: Remove

File: src/app/(dashboard)/attestation/utils/attestation-utils.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  signer
+}: CreateAttestationBodyDto): Promise<AttestationUuidDto> {
+  // Sepolia private data schema
+  // @TODO Implement a mapping to reference the correct private data schema for each network
+  const schemaUID =
+    "0x20351f973fdec1478924c89dfa533d8f872defa108d9c3c6512267d7e7e5dbc2";
+
+  const merkleTree = createMerkleTree(privateData);
+  const merkleRoot = merkleTree.root;
+  console.log(merkleTree);
```
Comment (AlecErasmus) [inline src/app/(dashboard)/attestation/utils/attestation-utils.ts]: Remove this

File: src/app/(dashboard)/attestation/utils/attestation-utils.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  });
+  const attestationUuid = await tx.wait();
+
+  return { attestationUuid: attestationUuid };
+}
+
+export function createMerkleTree(
+  privateData: AttestationPrivateDataDto,
+): StandardMerkleTree<EncodedMerkleValue> {
+  const salt =
```
Comment (AlecErasmus) [inline src/app/(dashboard)/attestation/utils/attestation-utils.ts]: Should pass this to save on backend, regenerate everything

File: src/app/(dashboard)/attestation/utils/attestation-utils.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      )
+    }
+  }
+  const merkleValues = encodeMerkleValues(valuesWithSalt);
+  const multiproof = merkleTree.getMultiProof(merkleValues);
+  const proofs = {
+    ...multiproof,
+    leaves: decodeMerkleValues(multiproof.leaves),
+  };
+  console.log(proofs);
```
Comment (AlecErasmus) [inline src/app/(dashboard)/attestation/utils/attestation-utils.ts]: Remember to remove

File: src/app/(dashboard)/attestation/utils/attestation-utils.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+
+function valueWithSalt(
+  field: string,
+  value: string,
+): MerkleValueWithSalt {
+  const salt =
+    "0xeba1f9c5ad55ba8569528641b3d105fb1ba09cf42b9918b9d535cebffaba8db4"; //FIXME
+
```
Comment (AlecErasmus) [inline src/app/(dashboard)/attestation/utils/attestation-utils.ts]: Don't for get

File: src/app/(dashboard)/attestation/attestationModal.tsx (hunk 1 above)
Comment (sudoFerraz) [reply src/app/(dashboard)/attestation/attestationModal.tsx:71]: This one I will keep for us to verify proofs until next PR 👍

**Review by AlecErasmus — APPROVED** [review]

**Pull request merged**